### PR TITLE
{bio}[intel/2019b] anvio v6.1 w/ Python 3.7.4, incl. scikit-learn 0.21.3, Biopython 1.75, Pysam 0.15.3

### PR DESCRIPTION
**Pull request by Darkless012**
(created using `eb --new-pr`)
requires ~~#9399~~

### CODE REVIEW

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        'checksums': ['033a11de5e3d19ea25c9302d11224e1a1898fe5abd23c61c7c360c25195e3eb1'],
+    }),
+    ('illumina-utils', '2.6', {
+        'modulename': 'IlluminaUtils',
+        'checksums': ['4ee7108d6ae67fc7d6c70bee4f775d38dfd921c10e4b020bd177838c649446ea'],
+    }),
+    ('more-itertools', '7.2.0', {
+        'checksums': ['409cd48d4db7052af495b09dec721011634af3753ae1ef92d2b32f73a745f832'],
+    }),
+
```
Comment (boegel) [inline easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb]: style nitpicking: please remove empty line

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    (name, version, {
+        'patches': ['%(name)s-%(version)s_requirements.patch'],
```
Comment (boegel) [inline easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb]: @Darkless012 I just realized it's a lot easier to change the `requirements.txt` at "runtime" as follows:

```python
(name, version, {
    'source_urls': ['https://github.com/%(github_account)s/%(name)s/releases/download/v%(version)s/'],
    'source_tmpl': '%(name)s-%(version)s.tar.gz',
    'checksums': ['a766514d47ba012b45fef51c0ad3a810f930687c6f59531f0d2e0bd96cb05db9'],
    # replace fixed (==) versions in requirements.txt with minimal versions (>=)
    'preinstallopts': "sed -i'' 's/==/>=/g' requirements.txt && ",
}),
```

That's one less patch file to worry about... (don't forget to remove the patch file too)

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+local_binaries_list = [
+    'anvi-pan-genome',
+    'anvi-script-reformat-fasta',
+    'anvi-profile',
+    'anvi-help',
+]
+
+sanity_check_paths = {
+    'files': ['bin/{}'.format(x) for x in local_binaries_list],
```
Comment (boegel) [inline easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb]: just for consistency with existing easyconfigs, please use `%s` instead:

```python
sanity_check_paths = {
    'files': ['bin/%s' % x for x in local_binaries_list],

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        'source_tmpl': '%(name)s-%(version)s.tar.gz',
+        'source_urls': ['https://github.com/%(github_account)s/%(name)s/releases/download/v%(version)s/'],
+        'checksums': [
+            'a766514d47ba012b45fef51c0ad3a810f930687c6f59531f0d2e0bd96cb05db9',  # anvio-6.1.tar.gz
+            'a34c4cde4d4027567fcb84e33299a17b11ab9c62106998a6f7364480ead3e6ec',  # anvio-6.1_requirements.patch
+        ],
+    }),
+]
+
+use_pip = True
```
Comment (boegel) [inline easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb]: style nitpicking: we usually put this above `exts_list`

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb]: I just tried this myself, and it made the installation fail, but for good reason...

`pip check` fails with:

```
anvio 6.1 has requirement scikit-learn>=0.21.2, but you have scikit-learn 0.20.3
```

You should make sure a sufficiently recent `scikit-learn` is used, certainly newer than what's specified in `requirements.txt`.

There's two ways to fix that:

1) drop the `scikit-learn` dependency and include the latest version in `exts_list`

2) update to `intel/2019b` and Python 3.7.4, and add easyconfigs for the missing dependencies (`Python`, `SciPy-bundle`, `matplotlib` and `h5py` are already there)

I strongly prefer option 2), that's definitely the best way forward here imho...

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+sanity_check_paths = {
+    'files': ['bin/{}'.format(x) for x in local_binaries_list],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+sanity_check_commands = [
+    'anvi-pan-genome --help',
+    'anvi-script-reformat-fasta --help',
+    'anvi-profile --version',
+    'anvi-help --help',
```
Comment (boegel) [inline easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb]: This is another good sanity check command (which goes beyond the ones already included): `anvi-self-test --suite mini`

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb (hunk 2 above)
Comment (Darkless012) [reply easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb]: Updating to 2019b

File: easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019b-Python-3.7.4.eb [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('SciPy-bundle', '2019.10', versionsuffix),
+    ('Pysam', '0.15.3'),
+    ('scikit-learn', '0.21.3', versionsuffix),
+    ('matplotlib', '3.1.1', versionsuffix),
+    ('prodigal', '2.6.3'),
+    ('Biopython', '1.75', versionsuffix),
+    ('h5py', '2.9.0', versionsuffix),
```
Comment (boegel) [inline easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019b-Python-3.7.4.eb]: @Darkless012 You'll need to bump this to `h5py` 2.10.0, see #9399...

**Comment by boegel** [comment]
@Darkless012 `anvi-self-test --suite mini` fails with:

```
Config Error: Well, in order to run this program, you need hmmscan to be installed on your
              system.
```

Looks like `HMMER` needs to be added to `dependencies` as well, which means another easyconfig (the Intel equivalent of `HMMER-3.2.1-gompi-2019b.eb`)...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in this PR)
node2412.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/48ce3e6bece519dc995e0aae15b9b1ab for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in this PR)
node3169.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/0ecf15e71554ff9a75918650736dd433 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Darkless012!

**Pull request merged**